### PR TITLE
FileTarget - MONO doesn't like using the native Win32 API

### PR DESCRIPTION
**Pull request by snakefoot**
Together with #1883, then it will also resolve the issue #1349 

This change is [Reviewable](https://reviewable.io/reviews/nlog/nlog/1893)

### CODE REVIEW

**Comment by 304NotModified** [comment]
thanks!!

File: src/NLog/Internal/FileAppenders/BaseFileAppender.cs [hunk 1]
```diff
@@ -290,7 +290,7 @@ private FileStream TryCreateFileStream(bool allowFileSharedWriting)
 #if !SILVERLIGHT && !MONO && !__IOS__ && !__ANDROID__
```
Comment (304NotModified) [inline src/NLog/Internal/FileAppenders/BaseFileAppender.cs]: should !MONO be removed?

Comment (snakefoot) [inline src/NLog/Internal/FileAppenders/BaseFileAppender.cs]: Well it saves some PlatformDetector-checks, when running pure MONO.

Comment (304NotModified) [inline src/NLog/Internal/FileAppenders/BaseFileAppender.cs]: true :)

**Pull request merged**